src/reflection.jl

```diff
@@ -336,7 +336,12 @@ function find_caller_of(interp::AbstractInterpreter, callee::MethodInstance, cal
 end
 
 function add_sourceline!(locs, CI, stmtidx::Int)
-    if isa(CI, IRCode)
+    if isdefined(CI, :debuginfo) # VERSION >= v"1.12"
+        stack = Base.IRShow.buildLineInfoNode(CI.debuginfo, :var"n/a", i)
```

Comment:
IIUC, the tests are failing because of this "n/a", which should instead be the MethodInstance object that is associated with this CI